src/dnaio/_core.pyx

```diff
@@ -384,15 +385,16 @@ def fastq_iter(file, sequence_class, Py_ssize_t buffer_size):
                 raise FastqFormatError(
                     "Length of sequence and qualities differ", line=n_records * 4 + 3)
 
-            ### Copy record into python variables
-            # PyUnicode_DecodeLatin1 is 50% faster than PyUnicode_DecodeASCII.
-            # This is because PyUnicode_DecodeLatin1 is an alias for
-            # _PyUnicode_FromUCS1. Which directly copies the bytes into a
-            # string object after some checks. With PyUnicode_DecodeASCII,
-            # there is an extra check whether characters exceed 128.
-            name = PyUnicode_DecodeLatin1(c_buf + name_start, name_length, 'strict')
-            sequence = PyUnicode_DecodeLatin1(c_buf + sequence_start, sequence_length, 'strict')
-            qualities = PyUnicode_DecodeLatin1(c_buf + qualities_start, qualities_length, 'strict')
+            # Constructing objects with PyUnicode_New and memcpy bypasses some of
+            # the checks otherwise done when using PyUnicode_DecodeLatin1 or similar
+            name = PyUnicode_New(name_length, 255)
```

Comment:
This will change the behaviour as previously PyUnicode_DecodeLatin1 would have found the highest character to be 127. So it would know the object to be of type `PyASCIIObject`. Now it will be another type.
I don't know if that is a very  significant change.

Reply:
Yes, I realize this changes the KIND. I think the difference is that a `PyASCIIObject` can safely return a pointer to its data when asked for the UTF-8 representation. If it’s just latin-1, the UTF-8 representation needs to be computed. But we never use the UTF-8 representation, right? Then there’s no difference in practice.